utils/compare_rule_var.py

```diff
@@ -0,0 +1,156 @@
+    returncode, stdout, stderr = run_command(["gh", "api", api_path])
+
+    if returncode != 0:
+        print(f"Exception: Could not fetch file. It may be new or deleted. "
+              f"(stderr: {stderr.strip()})")
+        return ""
+
+    try:
+        content_json = json.loads(stdout)
+        encoded_content = content_json.get('content', '')
+        if not encoded_content:
+            return ""
+
+        return base64.b64decode(encoded_content).decode('utf-8')
+    except (json.JSONDecodeError, KeyError, TypeError) as e:
+        print(f"Exception: Could not parse or decode content. Error: {e}", file=sys.stderr)
+        return ""
+
+
+def get_section_from_content(content: str, section: str) -> Optional[str]:
+    """
+    Extracts a section's value from a string containing the file content.
+    """
+    if not content:
+        return None
+
+    lines = content.splitlines()
+    found_ranges = find_section_lines(lines, section)
+
+    if found_ranges:
+        start, end = found_ranges[0]
+        section_content = lines[start: end + 1]
+        return '\n'.join(section_content)
+
+    return None
+
+
+def get_pr_shas(owner: str, repo: str, pr_number: int) -> Tuple[str, str]:
+    """
+    Fetches the base and head commit SHAs for a PR.
+    """
+    gh_pr_command = [
+        "gh", "pr", "view", str(pr_number),
+        "--repo", f"{owner}/{repo}",
+        "--json", "baseRefOid,headRefOid"
+    ]
+    returncode, stdout, stderr = run_command(gh_pr_command)
+
+    if returncode != 0:
+        raise RuntimeError(f"Failed to get PR details. GitHub CLI stderr: {stderr.strip()}")
+
+    try:
+        shas = json.loads(stdout)
+        base_sha = shas['baseRefOid']
+        head_sha = shas['headRefOid']
+        return base_sha, head_sha
+    except (json.JSONDecodeError, KeyError) as e:
+        raise ValueError(f"Could not parse commit SHAs from API response. Error: {e}") from e
+
+
+def get_value_from_commit(
+    owner: str,
+    repo: str,
+    file_path: str,
+    key: str,
+    sha: str
+) -> Optional[str]:
+    """
+    Fetches a file from a specific commit and extracts the value of a given key.
+
+    Args:
+        owner: The repository owner.
+        repo: The repository name.
+        file_path: The path to the file within the repository.
+        key: The section to extract from the file's content.
+        sha: The commit SHA from which to retrieve the file.
+
+    Returns:
+        The extracted value as a string, or None if the file or key is not found.
+    """
+    # 1. Fetch the file's content for the given commit SHA.
+    content = get_file_content_from_pr(owner, repo, sha, file_path)
+
+    # 2. Parse the content to get the desired value.
+    value = get_section_from_content(content, key)
+
+    return value
+
+
+def main():
+    """Main function to fetch PR files, parse them, and compare keys."""
+    parser = argparse.ArgumentParser(
```

Comment:
Most of the scripts in CaC/content use a separate function for arguments, similar to this: https://github.com/ComplianceAsCode/content/blob/master/utils/compare_versions.py#L223

I don't think this would block the PR, but it would be nice to also remove it from main to be more consistent with other scripts.